### PR TITLE
Various Mesh fixes

### PR DESCRIPTION
**Pull request by KKopyscinski**

### CODE REVIEW

**Comment by KKopyscinski** [comment]
Results of Mesh tests with these patches:

<details>

  1/227   MESH    MESH/NODE/PBADV/BV-01-C       PASS            64.082                  
  2/227   MESH    MESH/NODE/PBADV/BV-02-C       PASS            160.139                  
  3/227   MESH    MESH/NODE/PBADV/BV-03-C       PASS            27.5                  
  4/227   MESH    MESH/NODE/PBADV/BV-04-C       PASS            27.582                  
  5/227   MESH    MESH/NODE/PBADV/BI-01-C       PASS            62.303                  
  6/227   MESH    MESH/NODE/PBADV/BI-02-C       PASS            26.42                  
  7/227   MESH    MESH/NODE/PBADV/BI-03-C       PASS            58.035                  
  8/227   MESH    MESH/NODE/PBADV/BI-04-C       PASS            56.464                  
  9/227   MESH    MESH/NODE/MPS/BV-01-C         PASS            24.989                  
 10/227   MESH    MESH/NODE/MPS/BV-02-C         PASS            27.5                  
 11/227   MESH    MESH/NODE/MPS/BV-03-C         PASS            34.645                  
 12/227   MESH    MESH/NODE/MPS/BV-04-C         PASS            30.86                  
 13/227   MESH    MESH/NODE/MPS/BV-05-C         PASS            31.001                  
 14/227   MESH    MESH/NODE/MPS/BV-06-C         PASS            31.994                  
 15/227   MESH    MESH/NODE/MPS/BV-07-C         PASS            30.465                  
 16/227   MESH    MESH/NODE/MPS/BV-08-C         PASS            27.793                  
 17/227   MESH    MESH/NODE/MPS/BV-09-C         PASS            29.208                  
 18/227   MESH    MESH/NODE/MPS/BV-10-C         PASS            28.36                  
 19/227   MESH    MESH/NODE/MPS/BI-01-C         PASS            27.496                  
 20/227   MESH    MESH/NODE/PROV/BV-01-C        PASS            34.732                  
 21/227   MESH    MESH/NODE/PROV/BV-07-C        PASS            31.041                  
 22/227   MESH    MESH/NODE/PROV/BV-09-C        PASS            208.679                  
 23/227   MESH    MESH/NODE/PROV/BV-10-C        PASS            48.554                  
 24/227   MESH    MESH/NODE/PROV/BV-11-C        PASS            45.528                  
 25/227   MESH    MESH/NODE/PROV/BV-12-C        PASS            45.637                  
 26/227   MESH    MESH/NODE/PROV/BI-01-C        PASS            27.78                  
 27/227   MESH    MESH/NODE/PROV/BI-02-C        PASS            51.041                  
 28/227   MESH    MESH/NODE/PROV/BI-03-C        PASS            38.026                  
 29/227   MESH    MESH/NODE/PROV/BI-13-C        PASS            41.336                  
 30/227   MESH    MESH/NODE/PROV/BI-15-C        PASS            28.359                  
 31/227   MESH    MESH/NODE/BCN/SNB/BV-01-C     PASS            45.728                  
 32/227   MESH    MESH/NODE/NET/BV-01-C         PASS            35.322                  
 33/227   MESH    MESH/NODE/NET/BV-02-C         PASS            35.651                  
 34/227   MESH    MESH/NODE/NET/BV-03-C         PASS            34.464                  
 35/227   MESH    MESH/NODE/NET/BV-04-C         PASS            35.872                  
 36/227   MESH    MESH/NODE/NET/BV-05-C         PASS            36.301                  
 37/227   MESH    MESH/NODE/NET/BV-06-C         PASS            36.075                  
 38/227   MESH    MESH/NODE/NET/BV-07-C         PASS            34.361                  
 39/227   MESH    MESH/NODE/NET/BV-08-C         PASS            32.3                  
 40/227   MESH    MESH/NODE/NET/BV-09-C         PASS            32.053                  
 41/227   MESH    MESH/NODE/NET/BV-10-C         PASS            33.129                  
 42/227   MESH    MESH/NODE/NET/BV-11-C         PASS            32.459                  
 43/227   MESH    MESH/NODE/NET/BV-12-C         PASS            32.185                  
 44/227   MESH    MESH/NODE/NET/BV-13-C         PASS            33.193                  
 45/227   MESH    MESH/NODE/NET/BV-14-C         PASS            34.185                  
 46/227   MESH    MESH/NODE/NET/BI-01-C         PASS            33.973                  
 47/227   MESH    MESH/NODE/NET/BI-02-C         PASS            46.085                  
 48/227   MESH    MESH/NODE/NET/BI-03-C         PASS            31.583                  
 49/227   MESH    MESH/NODE/RLY/BV-01-C         PASS            62.2                  
 50/227   MESH    MESH/NODE/RLY/BV-02-C         PASS            100.736                  
 51/227   MESH    MESH/NODE/RLY/BI-01-C         PASS            34.945                  
 52/227   MESH    MESH/NODE/RLY/BI-02-C         PASS            41.493                  
 53/227   MESH    MESH/NODE/TNPT/BV-01-C        PASS            68.146                  
 54/227   MESH    MESH/NODE/TNPT/BV-02-C        PASS            69.724                  
 55/227   MESH    MESH/NODE/TNPT/BV-03-C        PASS            38.213                  
 56/227   MESH    MESH/NODE/TNPT/BV-04-C        PASS            66.597                  
 57/227   MESH    MESH/NODE/TNPT/BV-05-C        PASS            36.796                  
 58/227   MESH    MESH/NODE/TNPT/BV-06-C        PASS            34.266                  
 59/227   MESH    MESH/NODE/TNPT/BV-07-C        PASS            39.897                  
 60/227   MESH    MESH/NODE/TNPT/BV-08-C        PASS            47.624                  
 61/227   MESH    MESH/NODE/TNPT/BV-09-C        PASS            67.31                  
 62/227   MESH    MESH/NODE/TNPT/BV-10-C        PASS            35.867                  
 63/227   MESH    MESH/NODE/TNPT/BV-11-C        PASS            35.683                  
 64/227   MESH    MESH/NODE/TNPT/BV-12-C        PASS            125.009                  
 65/227   MESH    MESH/NODE/TNPT/BV-13-C        PASS            89.807                  
 66/227   MESH    MESH/NODE/TNPT/BI-01-C        PASS            66.872                  
 67/227   MESH    MESH/NODE/TNPT/BI-02-C        PASS            42.923                  
 68/227   MESH    MESH/NODE/IVU/BV-01-C         PASS            95.277                  
 69/227   MESH    MESH/NODE/IVU/BV-02-C         PASS            106.386                  
 70/227   MESH    MESH/NODE/IVU/BV-03-C         PASS            84.531                  
 71/227   MESH    MESH/NODE/IVU/BV-04-C         PASS            116.134                  
 72/227   MESH    MESH/NODE/IVU/BV-05-C         PASS            145.936                  
 73/227   MESH    MESH/NODE/IVU/BI-01-C         PASS            105.92                  
 74/227   MESH    MESH/NODE/IVU/BI-02-C         PASS            84.239                  
 75/227   MESH    MESH/NODE/IVU/BI-04-C         PASS            55.519                  
 76/227   MESH    MESH/NODE/IVU/BI-05-C         PASS            126.632                  
 77/227   MESH    MESH/NODE/IVU/BI-06-C         PASS            125.068                  
 78/227   MESH    MESH/NODE/KR/BV-01-C          PASS            128.833                  
 79/227   MESH    MESH/NODE/KR/BV-02-C          PASS            209.029                  
 80/227   MESH    MESH/NODE/KR/BV-03-C          PASS            77.244                  
 81/227   MESH    MESH/NODE/KR/BI-01-C          PASS            160.655                  
 82/227   MESH    MESH/NODE/KR/BI-02-C          PASS            84.339                  
 83/227   MESH    MESH/NODE/KR/BI-03-C          PASS            41.577                  
 84/227   MESH    MESH/NODE/KR/BI-04-C          PASS            106.15                  
 85/227   MESH    MESH/NODE/KR/BI-05-C          PASS            131.226                  
 86/227   MESH    MESH/NODE/FRND/FN/BV-01-C     PASS            52.883                  
 87/227   MESH    MESH/NODE/FRND/FN/BV-02-C     PASS            110.375                  
 88/227   MESH    MESH/NODE/FRND/FN/BV-03-C     PASS            51.865                  
 89/227   MESH    MESH/NODE/FRND/FN/BV-04-C     PASS            63.086                  
 90/227   MESH    MESH/NODE/FRND/FN/BV-05-C     PASS            54.236                  
 91/227   MESH    MESH/NODE/FRND/FN/BV-06-C     PASS            50.445                  
 92/227   MESH    MESH/NODE/FRND/FN/BV-07-C     PASS            40.793                  
 93/227   MESH    MESH/NODE/FRND/FN/BV-08-C     PASS            130.405                  
 94/227   MESH    MESH/NODE/FRND/FN/BV-09-C     PASS            124.493                  
 95/227   MESH    MESH/NODE/FRND/FN/BV-10-C     PASS            60.417                  
 96/227   MESH    MESH/NODE/FRND/FN/BV-11-C     PASS            39.65                  
 97/227   MESH    MESH/NODE/FRND/FN/BV-12-C     INCONC          58.447                  
 98/227   MESH    MESH/NODE/FRND/FN/BV-13-C     PASS            38.301                  
 99/227   MESH    MESH/NODE/FRND/FN/BV-14-C     PASS            126.337                  
100/227   MESH    MESH/NODE/FRND/FN/BV-15-C     PASS            96.084                  
101/227   MESH    MESH/NODE/FRND/FN/BV-16-C     PASS            62.732                  
102/227   MESH    MESH/NODE/FRND/FN/BV-17-C     PASS            54.666                  
103/227   MESH    MESH/NODE/FRND/FN/BV-18-C     PASS            51.462                  
104/227   MESH    MESH/NODE/FRND/FN/BV-19-C     PASS            65.179                  
105/227   MESH    MESH/NODE/FRND/FN/BV-20-C     PASS            78.274                  
106/227   MESH    MESH/NODE/FRND/FN/BV-21-C     PASS            41.463                  
107/227   MESH    MESH/NODE/FRND/FN/BV-22-C     PASS            65.577                  
108/227   MESH    MESH/NODE/FRND/FN/BV-23-C     PASS            34.826                  
109/227   MESH    MESH/NODE/FRND/FN/BI-01-C     PASS            42.49                  
110/227   MESH    MESH/NODE/FRND/FN/BI-02-C     PASS            51.226                  
111/227   MESH    MESH/NODE/FRND/FN/BI-03-C     PASS            57.107                  
112/227   MESH    MESH/NODE/FRND/LPN/BV-01-C    PASS            38.597                  
113/227   MESH    MESH/NODE/FRND/LPN/BV-02-C    PASS            129.992                  
114/227   MESH    MESH/NODE/FRND/LPN/BV-03-C    PASS            41.092                  
115/227   MESH    MESH/NODE/FRND/LPN/BV-04-C    PASS            41.075                  
116/227   MESH    MESH/NODE/FRND/LPN/BV-05-C    PASS            44.189                  
117/227   MESH    MESH/NODE/FRND/LPN/BV-06-C    PASS            41.844                  
118/227   MESH    MESH/NODE/FRND/LPN/BV-07-C    PASS            87.733                  
119/227   MESH    MESH/NODE/FRND/LPN/BV-08-C    PASS            40.204                  
120/227   MESH    MESH/NODE/FRND/LPN/BI-01-C    PASS            32.84                  
121/227   MESH    MESH/NODE/FRND/LPN/BI-02-C    PASS            40.073                  
122/227   MESH    MESH/NODE/FRND/LPN/BI-03-C    PASS            37.852                  
123/227   MESH    MESH/SR/PROX/BV-01-C          PASS            39.733                  
124/227   MESH    MESH/SR/PROX/BV-02-C          PASS            48.061                  
125/227   MESH    MESH/SR/PROX/BV-03-C          PASS            50.379                  
126/227   MESH    MESH/SR/PROX/BV-04-C          PASS            47.953                  
127/227   MESH    MESH/SR/PROX/BV-05-C          PASS            65.072                  
128/227   MESH    MESH/SR/PROX/BV-06-C          PASS            82.787                  
129/227   MESH    MESH/SR/PROX/BV-07-C          PASS            42.232                  
130/227   MESH    MESH/SR/PROX/BV-08-C          PASS            84.539                  
131/227   MESH    MESH/SR/PROX/BV-09-C          PASS            76.635                  
132/227   MESH    MESH/SR/PROX/BV-10-C          PASS            49.462                  
133/227   MESH    MESH/SR/PROX/BV-11-C          PASS            37.363                  
134/227   MESH    MESH/SR/PROX/BV-12-C          PASS            51.223                  
135/227   MESH    MESH/SR/PROX/BV-13-C          PASS            69.696                  
136/227   MESH    MESH/SR/PROX/BV-14-C          PASS            115.85                  
137/227   MESH    MESH/SR/PROX/BV-15-C          PASS            47.376                  
138/227   MESH    MESH/SR/PROX/BI-01-C          PASS            69.361                  
139/227   MESH    MESH/SR/PROX/BI-02-C          PASS            40.88                  
140/227   MESH    MESH/SR/MPXS/BV-01-C          PASS            39.088                  
141/227   MESH    MESH/SR/MPXS/BV-02-C          PASS            42.447                  
142/227   MESH    MESH/SR/MPXS/BV-03-C          PASS            41.007                  
143/227   MESH    MESH/SR/MPXS/BV-04-C          PASS            41.857                  
144/227   MESH    MESH/SR/MPXS/BV-05-C          PASS            39.361                  
145/227   MESH    MESH/SR/MPXS/BV-06-C          PASS            40.54                  
146/227   MESH    MESH/SR/MPXS/BV-07-C          PASS            37.99                  
147/227   MESH    MESH/SR/MPXS/BV-09-C          PASS            42.255                  
148/227   MESH    MESH/NODE/CFG/SNBP/BV-01-C    PASS            66.691                  
149/227   MESH    MESH/NODE/CFG/SNBP/BI-01-C    PASS            46.73                  
150/227   MESH    MESH/NODE/CFG/COMP/BV-01-C    PASS            34.736                  
151/227   MESH    MESH/NODE/CFG/DTTL/BV-01-C    PASS            35.67                  
152/227   MESH    MESH/NODE/CFG/DTTL/BI-01-C    PASS            43.844                  
153/227   MESH    MESH/NODE/CFG/GPXY/BV-01-C    PASS            37.821                  
154/227   MESH    MESH/NODE/CFG/GPXY/BV-02-C    PASS            45.535                  
155/227   MESH    MESH/NODE/CFG/GPXY/BI-01-C    PASS            35.848                  
156/227   MESH    MESH/NODE/CFG/CFGF/BV-01-C    PASS            36.073                  
157/227   MESH    MESH/NODE/CFG/CFGF/BI-01-C    PASS            37.338                  
158/227   MESH    MESH/NODE/CFG/CFGR/BV-01-C    PASS            37.159                  
159/227   MESH    MESH/NODE/CFG/CFGR/BI-01-C    PASS            37.981                  
160/227   MESH    MESH/NODE/CFG/MP/BV-01-C      PASS            39.017                  
161/227   MESH    MESH/NODE/CFG/MP/BV-02-C      PASS            45.98                  
162/227   MESH    MESH/NODE/CFG/MP/BI-01-C      PASS            38.06                  
163/227   MESH    MESH/NODE/CFG/MP/BI-02-C      PASS            38.368                  
164/227   MESH    MESH/NODE/CFG/MP/BI-03-C      PASS            40.011                  
165/227   MESH    MESH/NODE/CFG/SL/BV-01-C      PASS            40.172                  
166/227   MESH    MESH/NODE/CFG/SL/BV-02-C      PASS            37.498                  
167/227   MESH    MESH/NODE/CFG/SL/BV-03-C      PASS            35.836                  
168/227   MESH    MESH/NODE/CFG/SL/BI-01-C      PASS            34.963                  
169/227   MESH    MESH/NODE/CFG/SL/BI-02-C      PASS            37.735                  
170/227   MESH    MESH/NODE/CFG/SL/BI-03-C      PASS            37.107                  
171/227   MESH    MESH/NODE/CFG/SL/BI-04-C      PASS            34.616                  
172/227   MESH    MESH/NODE/CFG/SL/BI-05-C      PASS            33.121                  
173/227   MESH    MESH/NODE/CFG/SL/BI-06-C      PASS            34.323                  
174/227   MESH    MESH/NODE/CFG/SL/BI-07-C      PASS            37.931                  
175/227   MESH    MESH/NODE/CFG/SL/BI-08-C      PASS            34.197                  
176/227   MESH    MESH/NODE/CFG/SL/BI-09-C      PASS            34.612                  
177/227   MESH    MESH/NODE/CFG/SL/BI-10-C      PASS            36.19                  
178/227   MESH    MESH/NODE/CFG/NKL/BV-01-C     PASS            36.842                  
179/227   MESH    MESH/NODE/CFG/NKL/BV-02-C     PASS            34.952                  
180/227   MESH    MESH/NODE/CFG/NKL/BV-04-C     PASS            33.51                  
181/227   MESH    MESH/NODE/CFG/NKL/BI-01-C     PASS            36.705                  
182/227   MESH    MESH/NODE/CFG/NKL/BI-02-C     PASS            33.612                  
183/227   MESH    MESH/NODE/CFG/NKL/BI-03-C     PASS            36.632                  
184/227   MESH    MESH/NODE/CFG/AKL/BV-01-C     PASS            37.128                  
185/227   MESH    MESH/NODE/CFG/AKL/BV-02-C     PASS            39.284                  
186/227   MESH    MESH/NODE/CFG/AKL/BV-03-C     PASS            37.143                  
187/227   MESH    MESH/NODE/CFG/AKL/BI-01-C     PASS            34.992                  
188/227   MESH    MESH/NODE/CFG/AKL/BI-02-C     PASS            40.4                  
189/227   MESH    MESH/NODE/CFG/AKL/BI-03-C     PASS            33.897                  
190/227   MESH    MESH/NODE/CFG/MAKL/BV-01-C    PASS            36.604                  
191/227   MESH    MESH/NODE/CFG/MAKL/BV-02-C    PASS            36.391                  
192/227   MESH    MESH/NODE/CFG/MAKL/BI-01-C    PASS            35.494                  
193/227   MESH    MESH/NODE/CFG/MAKL/BI-02-C    PASS            36.198                  
194/227   MESH    MESH/NODE/CFG/MAKL/BI-03-C    PASS            34.147                  
195/227   MESH    MESH/NODE/CFG/MAKL/BI-04-C    PASS            35.115                  
196/227   MESH    MESH/NODE/CFG/NID/BV-01-C     PASS            35.192                  
197/227   MESH    MESH/NODE/CFG/NID/BV-02-C     PASS            146.862                  
198/227   MESH    MESH/NODE/CFG/NID/BV-03-C     PASS            36.641                  
199/227   MESH    MESH/NODE/CFG/NID/BI-01-C     PASS            55.335                  
200/227   MESH    MESH/NODE/CFG/NID/BI-03-C     PASS            55.864                  
201/227   MESH    MESH/NODE/CFG/RST/BV-01-C     PASS            35.051                  
202/227   MESH    MESH/NODE/CFG/HBP/BV-01-C     PASS            37.571                  
203/227   MESH    MESH/NODE/CFG/HBP/BV-02-C     PASS            63.732                  
204/227   MESH    MESH/NODE/CFG/HBP/BV-03-C     PASS            61.233                  
205/227   MESH    MESH/NODE/CFG/HBP/BV-04-C     PASS            49.165                  
206/227   MESH    MESH/NODE/CFG/HBP/BV-05-C     PASS            182.119                  
207/227   MESH    MESH/NODE/CFG/HBP/BV-06-C     PASS            40.789                  
208/227   MESH    MESH/NODE/CFG/HBP/BV-07-C     PASS            61.788                  
209/227   MESH    MESH/NODE/CFG/HBP/BI-01-C     PASS            49.426                  
210/227   MESH    MESH/NODE/CFG/HBP/BI-02-C     PASS            37.765                  
211/227   MESH    MESH/NODE/CFG/HBS/BV-01-C     PASS            38.097                  
212/227   MESH    MESH/NODE/CFG/HBS/BV-02-C     PASS            49.122                  
213/227   MESH    MESH/NODE/CFG/HBS/BV-03-C     PASS            49.027                  
214/227   MESH    MESH/NODE/CFG/HBS/BV-04-C     PASS            43.002                  
215/227   MESH    MESH/NODE/CFG/HBS/BV-05-C     PASS            83.952                  
216/227   MESH    MESH/NODE/CFG/HBS/BI-01-C     PASS            70.294                  
217/227   MESH    MESH/NODE/CFG/NTX/BV-01-C     PASS            33.899                  
218/227   MESH    MESH/NODE/CFG/LPNPT/BI-01-C   PASS            51.343                  
219/227   MESH    MESH/SR/HM/CFS/BV-01-C        PASS            69.725                  
220/227   MESH    MESH/SR/HM/CFS/BV-02-C        PASS            48.044                  
221/227   MESH    MESH/SR/HM/RFS/BV-01-C        PASS            36.294                  
222/227   MESH    MESH/SR/HM/RFS/BV-02-C        PASS            48.156                  
223/227   MESH    MESH/SR/HM/RFS/BV-03-C        PASS            45.715                  
224/227   MESH    MESH/SR/HM/RFS/BI-01-C        PASS            59.309                  
225/227   MESH    MESH/SR/HM/RFS/BI-02-C        PASS            57.977                  
226/227   MESH    MESH/SR/HM/HPS/BV-01-C        PASS            43.367                  
227/227   MESH    MESH/SR/HM/ATS/BV-01-C        PASS            40.61                  

Summary:

Status      Count
=================
INCONC          1
PASS          226
=================
Total         227
Regressions     0

</details>

Additionally, MESH/NODE/FRND/FN/BV-12-C was tested 5 more times and passed all of them, possible autopts issue. In conclusion, all tests are passing

**Review by michal-narajowski — CHANGES_REQUESTED** [review]
Commit titles should start with `mesh:` or `bttester:` and then capital letter. See commit history in nimble.

File: nimble/host/mesh/src/transport.c [hunk 1]
```diff
@@ -1413,8 +1413,9 @@ static int trans_seg(struct os_mbuf *buf, struct bt_mesh_net_rx *net_rx,
 	/* Reset the Incomplete Timer */
 	rx->last = k_uptime_get_32();
 
-	if (!k_delayed_work_remaining_get(&rx->ack) &&
-	    !bt_mesh_lpn_established()) {
+	if (!k_delayed_work_pending(&rx->ack) ||
```
[the remark this answers is not in the document]
Comment (KKopyscinski) [reply nimble/host/mesh/src/transport.c:1416]: Moved this commit to https://github.com/apache/mynewt-nimble/pull/978

**Comment by KKopyscinski** [comment]
@michal-narajowski Fixed commit messages and missing newlines at the end of files

File: nimble/host/mesh/src/transport.h [hunk 2]
```diff
@@ -106,8 +106,8 @@ struct bt_mesh_va *bt_mesh_va_get(uint16_t index);
 
 struct bt_mesh_va *bt_mesh_va_find(uint8_t uuid[16]);
 
-uint8_t bt_mesh_va_add(uint8_t uuid[16], uint16_t *addr);
+uint8_t bt_mesh_va_add(const uint8_t uuid[16], uint16_t *addr);
 
-uint8_t bt_mesh_va_del(uint8_t uuid[16], uint16_t *addr);
+uint8_t bt_mesh_va_del(const uint8_t uuid[16], uint16_t *addr);
 
 uint8_t *bt_mesh_va_label_get(uint16_t addr);
```
[the remark this answers is not in the document]
Comment (michal-narajowski) [reply nimble/host/mesh/src/transport.h:113]: Ping.

**Review by michal-narajowski — APPROVED** [review]

**Pull request merged**